core/src/test/java/com/google/errorprone/bugpatterns/ClassInitializationDeadlockTest.java

```diff
@@ -309,4 +310,63 @@ public void nonNestedSubclass() {
             "class B extends A {}")
         .doTest();
   }
+
+  @Test
+  public void simpleSubclassMethodReference() {
+    testHelper
+        .addSourceLines(
+            "Foo.java",
+            "import java.util.function.Supplier;",
+            "class A {",
+            "  static Supplier<B> supplier = B::new;",
+            "}",
+            "class B extends A {}")
+        .doTest();
+  }
+
+  @Test
+  public void compoundSubclassMethodReference() {
+    testHelper
+        .addSourceLines(
+            "Foo.java",
+            "import java.util.Comparator;",
+            "class A {",
+            "  static Comparator<B> comparator = Comparator.comparing(B::value);",
+            "}",
+            "class B extends A {",
+            "  int value;",
+            "  int value() {",
+            "    return value;",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void lambda() {
+    testHelper
+        .addSourceLines(
+            "Foo.java",
+            "import java.util.function.Supplier;",
+            "class A {",
+            "  static Supplier<B> supplier = () -> new B();",
+            "}",
+            "class B extends A {}")
+        .doTest();
+  }
+
+  @Test
+  public void subclassStaticMethod() {
+    testHelper
+        .addSourceLines(
+            "Foo.java",
+            "class A {",
+            "  // BUG: Diagnostic contains:",
+            "  static int value = B.value(); ",
+            "}",
+            "class B extends A {",
+            "  static int value() { return 0; }",
+            "}")
+        .doTest();
+  }
```

Comment:
The check's treatment of this situation is unchanged with this PR. I just noticed it wasn't tested and wanted to make sure it was still catching this usage while working on the check.